app/javascript/app/components/table.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Table } from 'react-bootstrap';
+
+function THeader(props) {
+  return (
+    <thead>
+      <tr>
+        {props.config.map((field, i) =>
+          <th key={'header-' + field.header}>
+            {field.header}
+          </th>
+        )}
+      </tr>
+    </thead>
+  );
+}
+
+function Row(props) {
+  return (
+    <tr key={'offer-' + props.offer_id + '-row'}>
+      {props.config.map((field, i) =>
+        <td key={'offer-' + props.offer_id + '-row-' + i}>
+          {field.data(props.offer)}
+        </td>
+      )}
+    </tr>
+  );
+}
+
+class AdminTable extends React.Component {
+  render() {
+    return (
+      <Table striped bordered condensed hover>
+        <THeader config={this.props.config} />
+        <tbody>
+          this.props.offers().map((offer) =>
```

Comment:
1) I just realized this should say `this.props.offers` rather than `this.props.offers()`. What's passed should be a list.
2) Yes this was meant to spread 
3) So in the config file, I'm assuming that I've retrieved the offers as JS objects after calling getOffers() from appState. If you look in appState you'll notice that when setting Offers (setOffers function) I don't change them to Immutable. :\ Still a tad bit conflicted with what the right thing to do is. Your input would be really appreciated.